### ISSUE TITLE
rubocop --auto-correct outputs invalid syntax

### ISSUE THREAD

**Issue by samphilipd** [bbatsov/rubocop#2924]
Running `rubocop --auto-correct` on real code converted valid syntax into invalid syntax.

Here is a test case, which is valid Ruby:

```ruby
if (not something.method_call 'arg' or
    not something_else.method_call 'arg' )
  do_something
end
```

Running `rubocop --auto-correct` on the above produces the following code, which will cause a syntax error:

```ruby
if !something.method_call 'arg' ||
   !something_else.method_call 'arg'
  do_something
end
```

## RuboCop version

```
$ rubocop -V
0.37.2 (using Parser 2.3.0.5, running on ruby 2.3.0 x86_64-darwin15)
```

**Comment by lumeet** [comment]
It seems to be fixed on master.

**Comment by alexdowad** [comment]
Yes, this is definitely fixed. Let's close this issue.

**Comment by samphilipd** [comment]
Great! Out of interest when is the next scheduled official release? (i.e. when will this fix be pushed to rubygems?)

**Comment by jonas054** [comment]
Fixed in `v0.38.0`.
